src/components/editor-content/editor-content.tsx

```diff
@@ -52,6 +85,184 @@ const EditorContent: React.FC<Props> = () => {
+      setSelectedStartingDate(date);
+    };
+
+    const handleStartWeekChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+      setStartWeek(event.target.value as Number);
+    };
+
+    if(scope.toString() !== "week"){
+      return (
+        <MuiPickersUtilsProvider utils={ DateFnsUtils } >
+          <Grid className={ classes.timeFilter }>
+            <KeyboardDatePicker
+              variant="inline"
+              views={ [scope] }
+              format={ dateFormat }
+              id="date-picker-start"
+              label={ strings.editorContent.filterStartingDate }
+              value={ selectedStartingDate }
+              onChange={ handleDateChange }
+              KeyboardButtonProps={ {'aria-label': `${ strings.editorContent.filterStartingDate }`} }
+            />
+          </Grid>
+        </MuiPickersUtilsProvider>
+      );
+    } else {
+      return (
+        <>
+          <FormControl variant="standard">
+            <MuiPickersUtilsProvider utils={ DateFnsUtils } >
+              <Grid className={ classes.timeFilterYearSelector }>
+                <KeyboardDatePicker
+                  variant="inline"
+                  views={ ["year"] }
+                  format="yyyy"
+                  id="date-picker-year-start"
+                  label={ strings.editorContent.selectYearStart }
+                  value={ selectedStartingDate }
+                  onChange={ handleDateChange }
+                  KeyboardButtonProps={ {'aria-label': `${ strings.editorContent.filterStartingDate }`} }
+                />
+              </Grid>
+            </MuiPickersUtilsProvider>
+          </FormControl>
+          <FormControl variant="standard" className={ classes.selectWeekNumbers }>
+            <FormHelperText>{ strings.editorContent.selectWeekStart }</FormHelperText>
+              <Select
+                labelId="scope-select-outlined-label"
+                id="scope-select-outlined"
+                value={ startWeek }
+                onChange={ handleStartWeekChange }
+              >
+                { generateWeekNumbers().map((weekNumber : number, index: number) => {
+                  return  <MenuItem 
+                            key={ index } 
+                            value={ weekNumber }
+                            >
+                            { weekNumber }
+                          </MenuItem>
+                }) }
+              </Select>
+          </FormControl>
+        </>
+      )
+    }
+  }
+
+  /**
+   * Renders ending datepicker/week selector depenging on scope
+   */
+  const renderEndDatePicker = () => {
+    const handleDateChange = (date: Date | null) => {
+      setSelectedEndingDate(date);
+    };
+    const handleEndWeekChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+      setEndWeek(event.target.value as Number);
+    };
+    const handleYearChange = (date: Date | null) => {
+      setSelectedEndingDate(date);
+    };
```

Comment:
Can be moved to outer scope.
And handleYearChange does not appear in the start date picker, is there any reason not using handleDateChange?

Reply:
Forgot to remove this, it's actually using handleDateChange. Will fix it.